Traceback (most recent call last):
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/kvarada/miniconda3/envs/jbook/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
import panel as pn
from panel import widgets
from panel.interact import interact
import matplotlib

pn.extension()

from sklearn.datasets import fetch_california_housing

cal = fetch_california_housing()
X_reg = cal.data
y_reg = cal.target

# Define a function to train a DecisionTreeRegressor with variable depth

def plot_depth_regression(depth):
    plt.clf()
    height_scale=5;width_scale=6
    reg_tree = DecisionTreeRegressor(max_depth=depth, random_state=42)
    reg_tree.fit(X_reg, y_reg)
    score = reg_tree.score(X_reg, y_reg)
    width = min(35, depth * 6)
    height = min(20, depth * 5 )
    fontsize = 11 - depth
    fig = plt.figure(figsize=(width, height))    
    plot_tree(reg_tree, feature_names=cal['feature_names'], fontsize = fontsize, filled=True)
    plt.title(f"Regression Tree Score: {score:.2f}", fontsize=14)        
    return fig

depth_selector = pn.widgets.IntSlider(name='depth', start=1, end=10, value=1)
pn.interact(plot_depth_regression, depth=depth_selector)
#pn.interact(f, depth=depth_selector).embed(max_opts=10)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mModuleNotFoundError[0m                       Traceback (most recent call last)
Cell [0;32mIn[25], line 1[0m
[0;32m----> 1[0m [38;5;28;01mimport[39;00m [38;5;21;01mpanel[39;00m [38;5;28;01mas[39;00m [38;5;21;01mpn[39;00m
[1;32m      2[0m [38;5;28;01mfrom[39;00m [38;5;21;01mpanel[39;00m [38;5;28;01mimport[39;00m widgets
[1;32m      3[0m [38;5;28;01mfrom[39;00m [38;5;21;01mpanel[39;00m[38;5;21;01m.[39;00m[38;5;21;01minteract[39;00m [38;5;28;01mimport[39;00m interact

[0;31mModuleNotFoundError[0m: No module named 'panel'

